shell_turtlestein.py

```diff
@@ -15,17 +20,17 @@ def cwd_for_window(window):
         3) The user's home directory.
     """
     folders = window.folders()
-    if len(folders) == 1:
-        return folders[0]
+    active_view = window.active_view()
+    active_file_name = active_view.file_name() if active_view else None
+    if not active_file_name:
+        return folders[0] if len(folders) else os.path.expanduser("~")
+    elif settings().get('active_file_folder'):
+        return os.path.dirname(active_file_name)
     else:
-        active_view = window.active_view()
-        active_file_name = active_view.file_name() if active_view else None
-        if not active_file_name:
-            return folders[0] if len(folders) else os.path.expanduser("~")
         for folder in folders:
             if active_file_name.startswith(folder):
                 return folder
-        return os.path.dirname(active_file_name)
+        return folders[0]
```

Comment:
I think we may have to handle the case where there aren't any open folders (probably just return the home dir in that case as well)